docs/architecture/adr-026-rosetta-api.md

```diff
@@ -0,0 +1,136 @@
+# ADR 26: Rosetta API Support
+
+## Changelog
+
+- August 3rd, 2020: Initial Draft.
+
+## Status
+
+- Approved, pending fine tuning the details of implementation.
+
+## Context
+
+We think it'd be greatly valuable to application developers to have the Cosmos SDK
+provide them with out-of-the-box Rosetta API support.
+According to [the project's website](https://www.rosetta-api.org/), Rosetta API is an open
+standard designed to simplify blockchain deployment and interaction. The latest specifications are
+available at [this URL](https://www.rosetta-api.org/docs/Reference.html).
+
+We want to achieve the following objectives:
+
+- Support multiple versions of Cosmos SDK.
+- Support Cosmos Hub.
+- Implement querying of historical data sets.
+
+## Decision
+
+We intend to develop a library that could be extended and used by application
```

Comment:
I'm a bit lost. if its a seperate repo why is there an ADR in the sdk repo?

Reply:
The idea was first to be part of the same sdk repo, but @zmanian pointed out that there is one use case where we need to be able to query multiple versions of the hub at the same time, so living in its own repo is more "easy" to maintain than to have the same code in multiple versions of the SDK. 

But in the end is very related to the SDK so I opened the ADR here. But if you think is not correct to keep it here we can change it.

Reply:
This ADR should be in the repo for where the code lives, IMO. Create repo -> add adr (design docs) to that repo -> code -> ship. What do others think?

Reply:
I think that is correct, meanwhile because the repo is not created yet (no permissions) we can use this as the draft discussion. Even I am not sure if finally everyone agrees on using another repo, this will be clarified in the cosmos-sdk call today I hope! Thanks @marbar3778